worker/sshserver/server.go

```diff
@@ -68,6 +77,19 @@ func NewServerWorker(config ServerWorkerConfig) (worker.Worker, error) {
 	s := &ServerWorker{config: config}
 	s.Server = &ssh.Server{
 		PublicKeyHandler: func(ctx ssh.Context, key ssh.PublicKey) bool {
+			if !names.IsValidUserName(ctx.User()) {
```

Comment:
a question: the ssh.Context holds all the information about the connection such as remote address, local address, user, client/server versions.. Should we not parse the remote address (the address the user is connecting to), to get the model UUID and then just fetch all ssh keys for that specific model and see if the specified `key` is in that set?